### PR TITLE
allow the SHA of the deployed image to be viewed

### PR DESCRIPTION
**Pull request by afeld**
Also refactored the deploy pipelines to all use a single file. Can't confirm the pipeline works until we merge, unfortunately.

### CODE REVIEW

**Review by thekaveman — APPROVED** [review]
Looks great.

Just want to confirm we are sure `${{ github.ref_name }}` contains only the branch name and nothing else? I remember looking through this a while ago and thought I read something about there being more like `refs/HEAD/{branch}`?? I could be thinking of something else.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -13,13 +15,16 @@ defaults:
 jobs:
   deploy:
     runs-on: ubuntu-latest
-    environment: dev
-    concurrency: dev
+    environment: ${{ github.ref_name }}
```
Comment (afeld) [inline .github/workflows/deploy.yml:18]: 90% sure `ref_name` is what we want.

Comment (afeld) [reply .github/workflows/deploy.yml:18]: https://github.com/github/docs/issues/15319 suggests it's correct. Let's find out.

Comment (afeld) [reply .github/workflows/deploy.yml:18]: Appears [it worked](https://github.com/cal-itp/benefits/runs/7047472242?check_suite_focus=true#step:6:9)!

Comment (thekaveman) [reply .github/workflows/deploy.yml:18]: Nice 👍 https://dev-benefits.calitp.org/static/sha.txt

**Pull request merged**